### DISCUSSION TITLE
Is there a way to remove the requirement for a space for extension shortcuts?

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [Ulauncher/Ulauncher#1419] [Q&A]
for example, I want emoji shortcuts to be `:`, but with no space. Today, I have to type `: ` (colon space) before typing reveals emojis.

**Comment by friday** [comment]
No, and we won't support that. #306 would be the way, eventually. We'd need regex to filter which extensions to use with what syntax. For example `1+1` detect as math, use calculator extension, `:` detect as start of an emoji, use emoji extension (only if the user opts in to use the emoji extension keywordless). `:` could mean other things too, and different users would want it for different things. So your emoji extension wouldn't have exlusive right to colon. There are other launchers that have reserved it for launcher internal things. catapult and pop-launcher I think both do that (ex type `:preferences` to open preferences), and I'm not making any promises that we won't do that.
